src/collection-view.js

```diff
@@ -277,13 +277,15 @@ const CollectionView = Backbone.View.extend({
   // Render children views.
   render() {
     if (this._isDestroyed) { return this; }
-    this.triggerMethod('before:render', this);
 
     this._destroyChildren();
 
     // After all children have been destroyed re-init the container
     this.children._init();
 
+    this.triggerMethod('before:render', this);
```

Comment:
well, 
in sources trigger `before:render` occurs before this._destroyChildren();
so, adding childView in before:render does nothing because they are effectivili destroyed by destroyChildren

Reply:
the same thing would happen if you showed something in a region in a view in `onBeforeRender` This is not an event for dealing with children.

Reply:
actually.. I think `dom:remove` will be ok.  Still part of the render is destroying the children.  this event should happen prior to that for that reason.